Mjml.Net/Components/Body/CarouselComponent.cs

```diff
@@ -0,0 +1,453 @@
+        public string Align = "center";
+
+        [Bind("border-radius", BindType.PixelsOrPercent)]
+        public string BorderRadius = "6px";
+
+        [Bind("container-background-color", BindType.Color)]
+        public string? ContainerBackgroundColor;
+
+        [Bind("icon-width", BindType.PixelsOrPercent)]
+        public string IconWidth = "44px";
+
+        [Bind("left-icon", BindType.String)]
+        public string LeftIcon = "https://i.imgur.com/xTh3hln.png";
+
+        [Bind("padding", BindType.PixelsOrPercent)]
+        public string? Padding;
+
+        [Bind("padding-bottom", BindType.PixelsOrPercent)]
+        public string? PaddingBottom;
+
+        [Bind("padding-left", BindType.PixelsOrPercent)]
+        public string? PaddingLeft;
+
+        [Bind("padding-right", BindType.PixelsOrPercent)]
+        public string? PaddingRight;
+
+        [Bind("padding-top", BindType.PixelsOrPercent)]
+        public string? PaddingTop;
+
+        [Bind("right-icon", BindType.String)]
+        public string RightIcon = "https://i.imgur.com/os7o9kz.png";
+
+        [Bind("tb-border", BindType.String)]
+        public string TbBorder = "2px solid transparent";
+
+        [Bind("tb-border-radius", BindType.PixelsOrPercent)]
+        public string TbBorderRadius = "6px";
+
+        [Bind("tb-hover-border-color", BindType.Color)]
+        public string TbHoverBorderColor = "#fead0d";
+
+        [Bind("tb-selected-border-color", BindType.Color)]
+        public string TbSelectedBorderColor = "#ccc";
+
+        [Bind("tb-width", BindType.PixelsOrPercent)]
+        public string? TbWidth;
+
+        [Bind("thumbnails", BindType.String)]
+        public string Thumbnails = "visible";
+
+        private string CarouselID { get; set; }
+
+        public IEnumerable<CarouselImageComponent> CarouselImages { get; private set; }
+
+        public (double widthValue, Unit widthUnit, string widthString, double ActualWidth) CurrentWidth { get; private set; }
+
+        public override void Measure(double parentWidth, int numSiblings, int numNonRawSiblings)
+        {
+            var widthValue = 0d;
+            var widthUnit = Unit.Pixels;
+            var widthString = string.Empty;
+
+            if (TbWidth != null)
+            {
+                (widthValue, widthUnit) = UnitParser.Parse(TbWidth);
+                widthString = TbWidth;
+            }
+            else
+            {
+                var percentWidthDouble = 1d / Math.Max(1, numNonRawSiblings);
+                widthValue = parentWidth * percentWidthDouble;
+                widthUnit = Unit.Pixels;
+                widthString = $"{widthValue}px";
+            }
+
+            CurrentWidth = (widthValue, widthUnit, widthString, ActualWidth);
+
+            // MeasureChildren(innerWidth);
+        }
+
+        public override void Render(IHtmlRenderer renderer, GlobalContext context)
+        {
+            CarouselID = Guid.NewGuid().ToString();
```

Comment:
@SebastianStehle, I'm using GUID but on the original it uses 6 random bytes. Should we change?

Reply:
No, it is fine, I have removed some attributes like "id" and "for" from the HTML diff.